pkg/detectors/aws/session_keys/sessionkey.go

```diff
@@ -161,14 +161,33 @@ func (s scanner) ShouldCleanResultsIrrespectiveOfConfiguration() bool {
 }
 
 const (
-	method   = "GET"
-	service  = "sts"
-	host     = "sts.amazonaws.com"
-	region   = "us-east-1"
-	endpoint = "https://sts.amazonaws.com"
+	method           = "GET"
+	service          = "sts"
+	regionCommercial = "us-east-1"
+	hostCommercial   = "sts.amazonaws.com"
+	regionGovCloud   = "us-gov-east-1"
+	hostGovCloud     = "sts." + regionGovCloud + ".amazonaws.com"
 )
 
-func (s scanner) verifyMatch(ctx context.Context, resIDMatch, resSecretMatch string, resSessionMatch string, retryOn403 bool) (bool, map[string]string, error) {
+func (s scanner) verifyMatch(ctx context.Context, resIDMatch, resSecretMatch, resSessionMatch string, retryOn403 bool) (bool, map[string]string, error) {
+	ok, data, err := s.attemptVerifyMatch(ctx, resIDMatch, resSecretMatch, resSessionMatch, retryOn403, regionCommercial, hostCommercial)
+	if err == nil && ok {
+		return true, data, nil
+	}
+
+	ok, data, err = s.attemptVerifyMatch(ctx, resIDMatch, resSecretMatch, resSessionMatch, retryOn403, regionGovCloud, hostGovCloud)
```

Comment:
As you mentioned, there isn’t a clear way to identify the specific endpoint. The changes look good to me overall.
My only concern is that we might end up attempting GovCloud verification for every error we receive from Commercial.
That said, I’m approving this from my side. If anyone has suggestions for a better approach, feel free to add.